File: 프로그래머스/월간 코드 챌린지 시즌1/이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

string toBin(int n){
    string str = "";
    while(n > 0){
        if(n % 2) str = "1"+str;
        else str = "0"+str;
        n /= 2;
    }
    
    return str;
}

vector<int> solution(string s) {
    int cnt=0, cnt_zero=0;
    while(s != "1"){
        // 0제거
        int len = s.length();
        for(int i=0; i<s.length(); i++){
            if(s[i] == '0'){
                cnt_zero++;
                len--;
            }
        }
        
        s = toBin(len);
        cnt++;
    }
    
    return {cnt, cnt_zero};
}

File: 프로그래머스/월간 코드 챌린지 시즌1/이진 변환 반복하기/이진 변환 반복하기.py
def toBin(n):
    str = ""
    while n > 0:
        if n%2 == 1: str = '1'+str
        else: str = '0'+str
        n //= 2

    return str

def solution(s):
    cnt,cnt_zero = 0,0
    while s != '1':
        l = len(s)
        for i in range(len(s)):
            if s[i] == '0':
                cnt_zero += 1
                l -= 1

        s = toBin(l)
        cnt += 1

    return [cnt, cnt_zero]
